### ISSUE TITLE
Unable to set `server_replay_ignore_params` in the UI

### ISSUE THREAD

**Issue by greatvovan** [mitmproxy/mitmproxy#4506]
#### Steps to reproduce the behavior:
1. Prepare a mock endpoint that works with 5 s delay. In this example I will use a configured mock https://hello.free.beeceptor.com/delay
2. Prepare a recording for the endpoint that contains calls to the endpoint with a parameter `param` (any value) and save it to `my.flow`.
3. Launch the software in the reverse proxy mode, using any of following:
  a. `mitmweb --mode reverse:https://hello.free.beeceptor.com --server-replay my.flow --server-replay-nopop --web-port 8888`
  b. `mitmproxy --mode reverse:https://hello.free.beeceptor.com --server-replay my_flow --server-replay-nopop`
4. In the UI:
  a. Go to Options, Edit Options, set `server_replay_ignore_params` to `param` and close Options.
  b. Press Shift-O for Options, scroll to `server_replay_ignore_params`, press `a`, enter `param`, press `Esc`, `q`, `q`.
5. Issue some requests through the proxy with value of `param` not existing in the recording. E.g.  
`curl 'http://localhost:8080/delay?param=value5'`

**EDIT**: corrected repro steps for TUI.

#### Problem
The setting has no effect, you will see 5 s delay in the response.

#### System Information
Mitmproxy: 6.0.2
Python:    3.9.2
OpenSSL:   OpenSSL 1.1.1i  8 Dec 2020
Platform:  macOS-10.15.7-x86_64-i386-64bit

**Comment by mhils** [comment]
Thanks! The problem here ist that we compute a hash for each flow when server replay is started, which we then later use to serve that flow:

https://github.com/mitmproxy/mitmproxy/blob/0650f132e9526a6ac881c2fb0cb3f53fba19ed1e/mitmproxy/addons/serverplayback.py#L114-L161

We need to rerun that function and recompute all hashes when one of the options affecting it changes. PRs welcome! 😃

**Comment by errorxyz** [comment]
Some notes that I found while testing this issue:
1. This issue only applies when `server_replay_ignore_params` is set through the UI after startup. It works as intended when the option is set before startup
i.e `mitmproxy --mode reverse:https://test-mitm.free.beeceptor.com --server-replay ~/my.flow --server-replay-reuse --set server_replay_ignore_params="param"` works as intended and ignores param while trying to match requests
2. If we set the option through the UI after startup, setting `server_replay_ignore_param` to `param=value` would match requests when the parameter for the request is `param=value` and return the matching response from the capture but wouldn't match `param=value5` i.e the entire `key=value` is matched when the option is set from the UI

I'll try to send a fix for this
